File: 1 - Two Sum/python/twosum.py
class Solution:
    def twoSum(self, nums, target):
        if len(nums) == 0 or nums == None:
            return []

        hashmap = {}

        for i, number in enumerate(nums):
            complement = target - number

            i2 = hashmap.get(complement, None)

            if i2 is not None:
                return [i2, i]

            hashmap[number] = i

        return []

File: 1 - Two Sum/ruby/twosum.rb
def two_sum(nums, target)
    return [] if nums.nil? or nums.empty? or target.nil?

    hashmap = {}

    nums.each_with_index do |number, i|
        complement = target - number

        i2 = hashmap.fetch(complement, nil)

        return [hashmap[complement], i] unless i2.nil?

        hashmap[number] = i
    end

    []
end
